### DISCUSSION TITLE
consuming-anonymous-types-with-diagnosticlistener-in-dotnet6/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#30] [Announcements]
# consuming-anonymous-types-with-diagnosticlistener-in-dotnet6/

In this post I look at how to consume DiagnosticListener events where the object is an anonymous type, and so can't be cast to a named Type.

https://andrewlock.net/consuming-anonymous-types-with-diagnosticlistener-in-dotnet6/

**Comment by CornedBee** [comment]
The easiest way to access properties of an unnamed object is to assign to a `dynamic`. No need to manually mess with reflection; the compiler will do it for you.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 21 2022, 06:38, in reply to [Thomas Levesque](https://disqus.com/by/thomaslevesque/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/30#discussioncomment-6063137)</em>

---

<p>Sure, you could. But personally I avoid dynamic wherever I can. This approach at least gives the best performance you can get</p>

[a comment by an automated account is not shown]
